### PR TITLE
drivers: support for Microchip MCP47xx DAC devices added

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR adds a driver which supports different Microchip MCP47xx DAC variants. Multiple MCP47xx DAC devices and different variants can be used at the same time.

Expander | Channels | Resolution
:--------|:--------:|:----------:
MCP4706  | 1        |  8-bit
MCP4716  | 1        | 10-bit
MCP4725  | 1        | 12-bit
MCP4726  | 1        | 12-bit
MCP4728  | 4        | 12-bit

The following features of MCP47xx DAC devices are not supported at the moment:
- configuring and reading address bits to/from EEPROM
- writing DAC channel output values to the EEPROM
- setting the UDAC bit and using the LDAC pin for MCP4728

The driver interface is kept as compatible as possible with the peripheral DAC interface. The only differences are that
- functions have the prefix `mcp47xx_` and
- functions require an additional parameter, the pointer to the MCP47xx device of type #mcp47xx_t.

### Testing procedure

Please refer the test application in `tests/driver_mcp47xx` for an example on how to use the driver
```
make -C tests/driver_mcp47xx BOARD=...
```

### CODE REVIEW

**Comment by gschorcht** [comment]
Still waiting for another PR.

File: drivers/include/mcp47xx.h [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    mcp47xx_pd_mode_t   pd_mode;    /**< Power-down mode selection */
+
+} mcp47xx_params_t;
+
+/**
+ * @brief   MCP47xx device data structure type
+ */
+typedef struct {
+    mcp47xx_params_t params;              /**< device configuration parameters */
+    uint16_t values[MCP47XX_CHN_NUM_MAX]; /**< contains the last values set */
```
Comment (benpicco) [inline drivers/include/mcp47xx.h:230]:
```suggestion
    uint16_t values[MCP47XX_CHN_NUM_MAX]; /**< contains the last values set for persistence when device is powered off  */
```

right?

File: drivers/include/mcp47xx.h [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+ * The value is always given as 16-bit value and is internally scaled to the
+ * actual resolution that the DAC unit provides, e.g., 12-bit.
+ *
+ * @param[in]   dev     descriptor of the MCP47xx DAC device
+ * @param[in]   chn     channel to set
+ * @param[out]  value   value to set line to
+ *
+ * @retval  none
+ */
+void mcp47xx_dac_get(mcp47xx_t *dev, dac_t chn, uint16_t *value);
```
Comment (benpicco) [inline drivers/include/mcp47xx.h:302]: I'm a bit surprised you implemented this when `periph/dac` API does not provide any `dac_get()`.
Feel free to keep it though :)

File: drivers/include/mcp47xx.h [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+} mcp47xx_t;
+
+#if IS_USED(MODULE_SAUL) || DOXYGEN
+/**
+ * @brief   MCP47xx configuration structure for mapping DAC channels to SAUL
+ */
+typedef struct {
+    const char *name;   /**< name of the MCP47xx device */
+    unsigned int dev;   /**< index of the MCP47xx device */
+    dac_t channel;      /**< channel of the MCP47xx device */
```
Comment (benpicco) [inline drivers/include/mcp47xx.h:240]: `dac_t` ?

**Review by benpicco — COMMENTED** [review]
Looks good! Only a few clarifications:

File: drivers/include/mcp47xx.h (hunk 3 above)
Comment (gschorcht) [reply drivers/include/mcp47xx.h:240]: Yes, it is the same as for the `perhiph/dac`. The goal was to keep it as compatible as possible with `periph/dac` and to handle channels in same way when we intended to introduced the DAC extension API with PR #10512.

File: drivers/include/mcp47xx.h (hunk 1 above)
Comment (gschorcht) [reply drivers/include/mcp47xx.h:230]: Yes, wil change it. May I squash directly?

Comment (benpicco) [reply drivers/include/mcp47xx.h:230]: Sure, squash

File: drivers/include/mcp47xx.h (hunk 3 above)
Comment (benpicco) [reply drivers/include/mcp47xx.h:240]: But `dac_t` would refer to a unique DAC device, not a channel of a DAC.
Just using `uint8_t` here would be less confusing IMHO (there also is no `saul_dac_params_t` yet)

Comment (gschorcht) [reply drivers/include/mcp47xx.h:240]: No, `dac_t` refers a line, that device + channel. It is also used in `periph/dac` to [set a certain channel of a device](https://doc.riot-os.org/group__drivers__periph__dac.html#ga179ae55de04d356cd6fbb63e9e3f51f6). In this sense, it also makes sense to use it here.

Comment (gschorcht) [reply drivers/include/mcp47xx.h:240]: When we started to introduce the extension API, we used exactly the same types for extenders to refer devices and channels.

Comment (gschorcht) [reply drivers/include/mcp47xx.h:240]: Unfortunatly, we had not much luck with the extension interfaces as suggested in PR #10512 thil now.

Comment (gschorcht) [reply drivers/include/mcp47xx.h:240]: However, we can of course also use `uint8_t`, if there is an extension API at some point in future, the type `dac_t` can still be mapped to `dev` + `chn`.

Comment (benpicco) [reply drivers/include/mcp47xx.h:240]: We can still change it later if we go forward with the extension API. :wink: 
To an unsuspecting user like me the `dac_t` type here just adds confusion.

**Comment by benpicco** [comment]
Please squash!

**Review by benpicco — APPROVED** [review]

**Pull request merged**